slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/dao/LowWatermarkDao.java

```diff
@@ -37,12 +37,13 @@ public void updateEpochWatermarksFor(
       final UInt64 targetEpoch) {
     handle
         .createUpdate(
-            "INSERT INTO low_watermarks (validator_id, source_epoch, target_epoch) VALUES (:validator_id, :source_epoch, :target_epoch) "
+            "INSERT INTO low_watermarks (validator_id, source_epoch, target_epoch) VALUES (:validator_id, :srcEpoch, :tgtEpoch) "
                 + "ON CONFLICT (validator_id) "
-                + "DO UPDATE set source_epoch = :source_epoch , target_epoch = :target_epoch")
+                + "DO UPDATE set source_epoch=:srcEpoch, target_epoch=:tgtEpoch WHERE (((low_watermarks.source_epoch <= :srcEpoch) and (low_watermarks.target_epoch <= :tgtEpoch)) OR "
```

Comment:
Is it possible for just one of the source or target epochs to be null? In that case what should happen?

Reply:
can't do it - we have an xnor constraint on the table